common/CMakeLists.txt

```diff
@@ -10,6 +10,9 @@ add_library(common_error STATIC
   error.cpp
   error/platform/c++11/error.cpp
 )
+# compile_options should be inherited by all our code, but common_error is used "virtually
+# everywhere", so just let it propagate from here.
```

Comment:
I am not convinced. My first grep for `target_link_libraries` got me [here](https://github.com/mendersoftware/mender/blob/01144d3d2a368c64a0f24b60a05c84f9ffac3b6a/api/CMakeLists.txt#L4), which does not use `common_error`.

Reply:
It does, but transitively, through the other dependencies.

But I see now that there are a few that don't, such as `common_setup` and `common`.

Update: I just discovered that `add_compile_options` works on "current directory and below`. This would be much better.

Reply:
@lluiscampos: This is completely changed now. The source was moved into the `src` folder, which allows us to use directory-scoped flags (like `add_compile_options`) again. So we don't need this hack anymore.

It was more logical to do the rename first, so the new commit is the first commit in the series, somewhat counter-intuitively. Then the rest of the changes are based on top of that, but are not significantly changed other than to take the `add_compile_options` opportunity into account.

Reply:
It looks much cleaner now :+1: